sharding/indexHashedNodesCoordinator.go

```diff
@@ -873,6 +922,53 @@ func (ihgs *indexHashedNodesCoordinator) ValidatorsWeights(validators []Validato
 	return weights, nil
 }
 
+func (ihgs *indexHashedNodesCoordinator) createActuallyLeavingPerShards(
+	unstakeLeaving map[uint32][]Validator,
+	additionalLeaving map[uint32][]Validator,
+	leaving []Validator,
+) (map[uint32][]Validator, map[uint32][]Validator) {
+	actuallyLeaving := make(map[uint32][]Validator)
+	actuallyRemaining := make(map[uint32][]Validator)
+
+	for shardId, leavingValidatorsPerShard := range unstakeLeaving {
+		sort.Sort(validatorList(leavingValidatorsPerShard))
```

Comment:
can't you use searchInMap - which might return shID as well - a lot of duplicated code.

Reply:
fixed